flik/linesearch/input.py

```diff
@@ -0,0 +1,51 @@
+""" Input processing of Line Search algorithm."""
+
+import numpy as np
+
+def check_input(var, func, grad, direction, alpha):
+    r"""Input control in Line Search algorithm
+
+    Parameters:
+    -----------
+    var: np.array((N,1), dtype=float)
+        Variables' current values
+    func: callable
+        Function :math:`f(x)`
+    grad: callable
+        Gradient of the function :math:`\nabla f(x)`
+    direction: np.ndarray((N,1)) for N variables
+        Direction vector for next step
+    alpha: float
+        Step length
+
+    Raises:
+    -------
+    TypeError:
+        If var is not a float or a 1-D np.ndarray
+        If func is not a callable
+        If grad is not a callable
+        If direction is not a float or a 1-D np.ndarray
+        If alpha is not a float
+    ValueError:
+        If var and direction don't have the same shape
+        If alpha is outside the interval (0., 1.]
+    """
+    if isinstance(var, float):
+        var = np.array(var)
+    elif not (isinstance(var, np.ndarray) and var.ndim == 1):
+        raise TypeError("Variable vector should be a float or a 1-D numpy.ndarray")
+    if not callable(func):
```

Comment:
could you add a newline here just to separate out different tests? I think it helps with readability